### ISSUE TITLE
MyLocationNewOverlay drawable not anchored correctly

### ISSUE THREAD

**Issue by mlendl** [osmdroid/osmdroid#1360]
Please note: issues related to or caused by osmbonuspack will be closed with a
suggestion to open an issue at https://github.com/MKergall/osmbonuspack

(feel free to delete whatever doesn't apply)

## Issue Type

[ ] Question
[x] Bug
[ ] Improvement
[ ] Build system related
[ ] Performance
[ ] Documentation

## Description and/or steps/code to reproduce the problem
I am not 100% sure, but it seems that the drawables of current location are not correctly anchored. I'd expect that person should be anchored in bottom center and arrow in center. Btw the current location seems to be positioned correctly in the center of MapView. Please check it.

### If it's a bug, version(s) of android this affects:

All

### Version of osmdroid the issue relates to:

6.1.0

**Comment by spyhunter99** [comment]
can you provide code and/or steps to reproduce?

**Comment by mlendl** [comment]
I am providing two screen shots. On the first one there is my location
correctly (responding the reality) in the center given by the cross. On the
second one there is the same location with the person icon, where is
visible that the person is not positioned correctly..

Corresponding code:

  mMyLocationOverlay = new MyLocationNewOverlay(mMapView);
  BitmapDrawable stop = (BitmapDrawable)
ResourcesCompat.getDrawable(getResources(), R.drawable.location_stop,
null);
  BitmapDrawable arrow = (BitmapDrawable)
ResourcesCompat.getDrawable(getResources(), R.drawable.location_arrow,
null);
  mMyLocationOverlay.setDirectionArrow(stop.getBitmap(), arrow.getBitmap());
  mMyLocationOverlay.setDrawAccuracyEnabled(true);
  mMyLocationOverlay.runOnFirstFix(() -> {
    TrackUtils.playTone(ToneGenerator.TONE_CDMA_EMERGENCY_RINGBACK);
    GeoPoint myLocation = mMyLocationOverlay.getMyLocation();
    if (myLocation != null) {
      runOnUiThread(() -> mMapView.getController().animateTo(myLocation));
    }
  });
}

**Comment by mlendl** [comment]
the screenshots:

![device-2019-12-14-194448](https://user-images.githubusercontent.com/9496205/71001447-9a7d8500-20dd-11ea-8e39-24b9d199dc0a.png)
![device-2019-12-14-194539](https://user-images.githubusercontent.com/9496205/71001448-9a7d8500-20dd-11ea-84eb-a31ef23a4387.png)

**Comment by spyhunter99** [comment]
try calling setPersonHotspot....that's the anchor point for the icon. i.e. the position on the marker that you want to represent the location on the map

**Comment by mlendl** [comment]
not very clear how to set the x, y params, probably a complex calculation
needed according to icon resolution and map scale
couldn't it be resolved somehow defaultly, because it must be common
problem regardless the customized icons

**Comment by monsieurtanuki** [comment]
@mlendl Indeed, [line 113 of MyLocationNewOverlay](https://github.com/osmdroid/osmdroid/blob/master/osmdroid-android/src/main/java/org/osmdroid/views/overlay/mylocation/MyLocationNewOverlay.java#L113) stinks:
````java
mPersonHotspot = new PointF(24.0f * mScale + 0.5f, 39.0f * mScale + 0.5f);
````

It should be more like what happens in [Marker](https://github.com/osmdroid/osmdroid/blob/master/osmdroid-android/src/main/java/org/osmdroid/views/overlay/Marker.java#L523), where the anchor has a horizontal and a vertical [0,1] factor, to be multiplied by the icon's width and height.

**Comment by mlendl** [comment]
sorry guys, still not clear the magic numbers, but probably I would be able to implement some correction, but I am not sure if it helps, because the icon is positioned completly out of the location gray circle, I think the solution should be within Osmdroid, because the problem is probably general concernig all users

**Comment by monsieurtanuki** [comment]
@mlendl I've just PR'ed #1460 - can you try again with this version?

**Comment by mlendl** [comment]
sorry I don't know to checkout the branch, but actually I have customized
only the direction icon and everything's OK

**Comment by zamojski** [comment]
@spyhunter99 When can we expect this to be released? I have [the same issue reported by my users](https://github.com/zamojski/TowerCollector/issues/101).

@mlendl How did you resolved it?

**Comment by AndroidPat** [comment]
Not sure if it's the same issue but when I use a custom drawable (Google Maps-like circle) with setPersonIcon(), the position of the icon changes in relation to the map when I zoom out placing the user in the incorrect location. This does not happen when I use the default yellow guy.

**Comment by zamojski** [comment]
I had exactly the same observation, this is the same issue.

**Comment by monsieurtanuki** [comment]
@zamojski @AndroidPat Does #1460 fix your issue? Even without connecting to that branch and before a merge and a release, you can just duplicate and rename class `MyLocationNewOverlay` (from #1460) and test it in your code.
I hope this works. Please tell us if it does.

**Comment by zamojski** [comment]
@monsieurtanuki  I think it does what we need. The change in commit 53bd75267b368f18971b4dc6e25eea83987b3679 looks similar to what I have in re-implemented `setDirectionArrow` method. I've also checked the class itself in my project as you suggested and it looks ok.

**Comment by monsieurtanuki** [comment]
@spyhunter99 I guess #1460 could be part of the next release.

**Comment by nazmos-sakib** [comment]
i had the same issue.
`        myLocationOverlay.setPersonAnchor(.5f,.5f);`
adding this after _`setPersonIcon()`_ function call, fixed the problem.

in the documentation it states 
_

> Anchors for the person icon Expected values between 0 and 1, 0 being top/left, .5 center and 1 bottom/right
> Since:
> 6.2.0

_

**Comment by AlexSuvorov2k** [comment]
Yes, order of calling methods is important:
```
val mLocationOverlay = MyLocationNewOverlay(GpsMyLocationProvider(context), binding.mapView)
mLocationOverlay.setPersonIcon(userIcon.toBitmap())
mLocationOverlay.setPersonAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_CENTER)
mLocationOverlay.setDirectionIcon(userArrow.toBitmap())
mLocationOverlay.setDirectionAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_CENTER)
binding.mapView.overlays.add(mLocationOverlay)
```
